subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -280,7 +300,7 @@ static bool validate_firmware(u32_t fw_dst_address, u32_t fw_src_address,
 	}
 
 	fw_val_info = validation_info_find(
-		fw_src_address + fwinfo->size, 4);
+		fw_src_address + fwinfo->size, 4, fwinfo);
```

Comment:
1) with the changed API, the code always passes the `fwinfo` struct to `validation_info_find()`, so why not let the function itself add `+ fwinfo->size` internally ?

2) `4` is magic number.

Reply:
1. the argument is not always used like this, see the other call.

2. I can change it to 0, indicating that we know exactly where the data is.

Reply:
>     1. the argument is not always used like this, see the other call.
 
What I see is two calls:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/f4d6b6225d2dce5cb6744bf933f9a14f9693993b/subsys/bootloader/bl_validation/bl_validation.c#L151-L154
where `next_info->size` is used and `next_info` struct itself is given as argument to the function.
and
https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/f4d6b6225d2dce5cb6744bf933f9a14f9693993b/subsys/bootloader/bl_validation/bl_validation.c#L302-L303
where `fw_info->size` is used and `fw_info` struct itself is also given to the function.

so in both cases it seems the `size` member in the struct following is added to source address.
So why is it not the function that adds the size member from the struct to the address given.

>     2. I can change it to 0, indicating that we know exactly where the data is.

What I meant, is that `4` is magic to me. What does 4 mean ?
It should have a define somewhere:

```suggestion
      fw_src_address + fwinfo->size, MY_UNEXPLAINED_VALUE_OF_FOUR, fwinfo);
```

Reply:
#define SEARCH_DISTANCE 4 ?

Reply:
> and
> https://github.com/NordicPlayground/fw-nrfconnect-nrf/blob/f4d6b6225d2dce5cb6744bf933f9a14f9693993b/subsys/bootloader/bl_validation/bl_validation.c#L302-L303
> 
> 
> where `fw_info->size` is used and `fw_info` struct itself is also given to the function.
> so in both cases it seems the `size` member in the struct following is added to source address.
> So why is it not the function that adds the size member from the struct to the address given.

In the second case it is not used. Its the fw_src_address, and this is not from fw_info.

Reply:
The magic value is not something introduced in this PR, but I will fix it anyhow :D

Reply:
> In the second case it is not used. Its the fw_src_address, and this is not from fw_info.

seems we are talking about two different things:
What i'm suggesting is to change:
```
fw_val_info = validation_info_find(fw_src_address + fwinfo->size, 4, fwinfo);
```
into:
```
fw_val_info = validation_info_find(fw_src_address, 4, fwinfo);
```
and
```
validation_info_find(next_info->address + next_info->size, 4, next_info);
```
into
```
validation_info_find(next_info->address, 4, next_info);
```
and then inside the `validation_info_find()` you add the `start_address += finfo->size` 
So that `validation_info_find()` itself adds the size value of the struct provided.

But this is really just a minor suggestion, just fought it would be nice to keep this knowledge inside the function, as the info is already available in the struct.

Reply:
> The magic value is not something introduced in this PR, but I will fix it anyhow :D

That's the danger of touching such a line.
It drags attention :laughing:

Reply:
Haha! Good point! Next time I will dance around stuff like this.

I get your point now, sorry for being a bit slow. I will see if this change is an improvement.